### PR TITLE
Generate code to notify error for streaming operations

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
Generating code to notify error for streaming operations.

```
            executeFuture.whenComplete((r, e) -> {
                if (e != null) {
                    runAndLogError(log, "Exception thrown in exceptionOccurred callback, ignoring",
                            () -> asyncResponseTransformer.exceptionOccurred(e));
                }
            });
```
Per docs,
https://github.com/aws/aws-sdk-java-v2/blob/5a799393ce9f34b07625ca5d614e5f6367ec7780/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncRetryableStage.java#L167-L170

we should invoke `AsyncResponseTransformer#exceptionOccurred` in the generated client code; we were only generating the code in json services previously, so the fix is to add the generated code for xml services.
 

## Testing
added codegen tests, unit tests, integ tests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-query-async-client-class.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+                new ClientExecutionParams<StreamingOutputOperationRequest, StreamingOutputOperationResponse>()
+                    .withOperationName("StreamingOutputOperation")
+                    .withMarshaller(new StreamingOutputOperationRequestMarshaller(protocolFactory))
+                    .withResponseHandler(responseHandler).withErrorResponseHandler(errorResponseHandler)
+                    .withInput(streamingOutputOperationRequest), asyncResponseTransformer);
+            executeFuture.whenComplete((r, e) -> {
+                if (e != null) {
+                    asyncResponseTransformer.exceptionOccurred(e);
+                }
+            });
```
Comment (millems) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-query-async-client-class.java]: should we return the result of executeFuture.whenComplete? I think the only difference is that if exceptionOccurred throws an exception, the exception is propagated to the result future.

Comment (zoewangg) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-query-async-client-class.java]: Maybe we should do `runAndLogError` like line 252. Customers probably don't care if exception occurs in `exceptionOccurred`.

Comment (millems) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-query-async-client-class.java]: Sure! I don't think exceptionOccurred is supposed to raise an exception anyway, right?

Comment (zoewangg) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-query-async-client-class.java]: Right, at least not in our built-in `AsyncResponseTransformer`s. I'll update the javadoc for `exceptionOccurred` to make it more clearly.

**Pull request merged**